docs/local_source_upload.md

```diff
@@ -0,0 +1,44 @@
+Local Source Upload
+-------------------
+
+The purpose of project Shipwright is to transform source code into container images, using known strategies to accomplish this task.
+
+Now, with local source upload, we move Shipwright closer to the developer's inner loop, by allowing streaming local content to the Build Controller, running in a Kubernetes cluster, and creating a container image out of if. Users can try out Shipwright before submitting a pull-request, and rely on the cluster's computing power.
+
+# Usage
+
+The following steps represents the traditional workflow using local source upload:
+
+1. Create a Build or use a pre-existing one. On the Build resource we register the common settings, like the Build Strategy. For instance:
+
+```bash
+shp build create sample-nodejs \
+    --source-url="https://github.com/shipwright-io/sample-nodejs.git" \
+    --output-image="docker.io/<namespace>/sample-nodejs:latest"
+```
+
+2. Clone the repository, or use a pre-existing one. For instance:
+
+```bash
+git clone https://github.com/shipwright-io/sample-nodejs.git && \
+    cd sample-nodejs
+```
+
+3. Keep working on the project, when you're ready to build a new container image with the local changes, run:
+
+```bash
+shp build upload sample-nodejs --follow \
+    --output-image="docker.io/<namespace>/sample-nodejs:<tag>"
+```
+
+4. Rinse and repeat the previous step
+
+Notice we have a few placeholders delimited by `<>`, you should use a regular Container Registry where the Shipwright can push the images.
+
+# Streaming
+
+The subcommand `build upload` creates a new `BuildRun` for the informed `Build`. The newly created BuildRun contains settings to instruct the Build Controller to wait for local user upload, instead of cloning the external repository as usual.
```

Comment:
```suggestion
The subcommand `build upload` creates a new `BuildRun` for the informed `Build`. The newly created `BuildRun` contains settings to instruct the Build Controller to wait for the local user upload instead of cloning the external repository as usual.
```